walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,100 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
+import java.util.Stack;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Cases
+    // Options for no attendees - should be available all day.
+    if (request.getAttendees().isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    // Duration longer than a day will return empty array.
+    if (request.getDuration() > TimeRange.getTimeInMinutes(23, 59)) {
+      return Arrays.asList();
+    }
+    // Check if there are availabilities for all attendees
+    List<String> attendees = new ArrayList<>();
+    attendees.addAll(request.getAttendees());
+    attendees.addAll(request.getOptionalAttendees());
```

Comment:
I have checked, you can ignore all optional attendees here and pass all the tests. (I think it is what is meant to do here, rather than considering all mandatory attendees and optional attendees the same directly.)
However, if you will make changes on the attendees in the next step, feel free to have no changes here.

Reply:
Yeah I will be adding tests to check optional vs mandatory attendees afterwards :) But thanks for letting me know!